src/main/java/com/microsoft/azure/functions/worker/broker/JavaFunctionBroker.java

```diff
@@ -63,9 +64,12 @@ public void loadMethod(FunctionMethodDescriptor descriptor, Map<String, BindingI
 	}
 
 	private void initializeOneTimeLogics() {
```

Comment:
Hi @trask , I remember when I first implement this method I am using below code

```java
private void initializeOneTimeLogics() {
	if (!oneTimeLogicInitialized) {
		synchronized (lock) {
			if (!oneTimeLogicInitialized) {
				initializeInvocationChainFactory();
				initializeFunctionInstanceInjector();
				oneTimeLogicInitialized = true;
			}
		}
	}
}
```

I am using to if condition here (the idea is for performance improving), but this code is exact same as we use `AtomicBoolean`, it will still have this NPE issue, as we can skip the core initialize logics here. So with/without using `AtomicBoolean` we will get this issue anyway :)

Reply:
I don't think the above code would suffer the NPE, the second thread will block on the synchronized since you don't update the boolean until the end